controllers/machine_controller.go

```diff
@@ -242,6 +255,73 @@ func (r *MachineReconciler) isDeleteNodeAllowed(ctx context.Context, machine *cl
 	}
 }
 
+func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
+	var kubeClient kubernetes.Interface
+	if cluster == nil {
+		var err error
+		kubeClient, err = kubernetes.NewForConfig(r.config)
+		if err != nil {
+			return fmt.Errorf("unable to build kube client: %v", err)
+		}
+	} else {
+		// Otherwise, proceed to get the remote cluster client and get the Node.
+		remoteClient, err := remote.NewClusterClient(r.Client, cluster)
+		if err != nil {
+			klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+				cluster.Name, name, err)
+			return nil
+		}
+		var err2 error
+		kubeClient, err2 = kubernetes.NewForConfig(remoteClient.RESTConfig())
+		if err2 != nil {
+			klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
```

Comment:
Do we want to record an event here against the Machine so the user is informed?

Reply:
We don't do that for the call to deleteNode(), and the code will fall through to that point, so we can just add an event there.  As a follow-on, perhaps we should generate the client before both of these functions and pass it in.

Reply:
Thanks for pointing out the consistency with drainNode. I had missed that. +1 to possibly making changes in a follow-up.